WangscapeTest/TestCoordinatePacker.cpp

```diff
@@ -0,0 +1,71 @@
+#include <gtest/gtest.h>
+
+#include <vector>
+#include <algorithm>
+
+#include <tilegen/CoordinatePacker.h>
+
+class TestCoordinatePacker : public ::testing::Test
+{
+protected:
+};
+
+TEST_F(TestCoordinatePacker, TestCoordinatePackerDomain)
+{
+    for (auto i : {0, 1, 2, 10})
+    {
+        tilegen::CoordinatePacker<size_t> coord_packer(i);
+        EXPECT_ANY_THROW(coord_packer.addCoordinate(-1));
+        EXPECT_ANY_THROW(coord_packer.addCoordinate(i));
+        EXPECT_ANY_THROW(coord_packer.addCoordinate(i + 1));
+    }
+}
+TEST_F(TestCoordinatePacker, TestCoordinatePackerSize)
+{
+    for (auto i : {0, 1, 2, 10})
+    {
+        tilegen::CoordinatePacker<size_t> coord_packer(i);
+        EXPECT_EQ(i, coord_packer.size());
+    }
+}
+
+TEST_F(TestCoordinatePacker, TestCoordinatePacker1)
```

Comment:
Name `TestCoordinatePacker1` doesn't tell very much about this testcase. Could you find a better name?